trpcultivate_phenotypes/tests/src/Kernel/Display/DisplayValidationResultWindowTest.php

```diff
@@ -0,0 +1,406 @@
+            'details' => [],
+          ],
+        ],
+        [
+          'tcp-validate-pass',
+          'tcp-validate-fail',
+          'tcp-validate-todo',
+        ],
+      ],
+      // #10: Malformed render array.
+      [
+        'malformed render array',
+        [
+          [
+            'title' => 'Validation Title Text - Fail',
+            'status' => 'fail',
+            'details' => [
+              '#not_my_type' => 'not_a_type',
+              '#not_a_property' => 'some property',
+              '#markup' => 'Lorem ipsum dolor sit amet',
+              'table' => [
+                '#type' => 'table',
+                '#captionzzzz' => 'Validator Case Message',
+                '#header-array' => ['Header 1', 'Header 2'],
+                '#row' => [
+                  ['Row #1 - Value #1', 'Row #1 - Value #2', 'Row #1 - Value #3', 'Row #1 - Value #4'],
+                ],
+              ],
+            ],
+          ],
+        ],
+        ['tcp-validate-fail'],
+      ],
+    ];
+  }
+
+  /**
+   * Test result window display.
+   *
+   * @dataProvider provideValidationresultRenderArray
+   */
+  public function testResultWindowDisplay($scenario, $validation_result_input, $expected_class) {
+
+    $validation_window = [
+      '#type' => 'inline_template',
+      '#theme' => 'result_window',
+      '#data' => [
+        'validation_result' => $validation_result_input,
+      ],
+    ];
+
+    // This is now the validation result window markup with all validation
+    // result details rendered.
+    $validation_window_markup = $this->renderer->renderRoot($validation_window);
+
+    // For every validation item in a test scenario, test that:
+    // 1. The validation item has been set the correct status-based class name.
+    // 2. The validation item has the correct title text.
+    // 3. The details render array rendered markup is present in the overall
+    // markup output.
+    foreach ($validation_result_input as $i => $validation_item) {
+      $class_name = $expected_class[$i];
+      $this->assertStringContainsString(
+        $class_name,
+        $validation_window_markup,
+        'The class name ' . $class_name . ' of validation input #' . $i . ' in scenario ' . $scenario . ', was not found in the validation window markup'
+      );
+
+      $title_text = $validation_item['title'];
+      $this->assertStringContainsString(
+        $title_text,
+        $validation_window_markup,
+        'The title text ' . $title_text . ' of validation input #' . $i . ' in scenario ' . $scenario . ', was not found in the validation window markup'
+      );
+
+      $details_markup = $this->renderer->renderRoot($validation_item['details']);
+      $this->assertStringContainsString(
+        $details_markup,
+        $validation_window_markup,
+        'The details markup of validation input #' . $i . ' in scenario ' . $scenario . ', was not found in the validation window markup'
+      );
+    }
+  }
```

Comment:
This section of asserts is only testing that the parts of a single validation item are SOMEWHERE in the validation window, not that they are where they are supposed to be. For example, if you have two validations items where the first one passes and the second fails, this only checks that there is a pass class somewhere and a fail class somewhere. This means the test would still pass if they were switched or if the surrounding markup was mangled.

Instead we would want to ensure that the first item contains the elements the first item should with the wrapper markup that we expect.